devito/finite_differences/differentiable.py

```diff
@@ -0,0 +1,182 @@
+import sympy
+from sympy.core.basic import _aresame
+import numpy as np
+
+from devito.symbolics.extended_sympy import ExprDiv
+
+__all__ = ['Differentiable']
+
+
+class Differentiable(sympy.Expr):
+    """
+    This class represents Devito differentiable objects such as
+    sum of functions, product of function or FD approximation and
+    provides FD shortcuts for such expressions
+    """
+    _op_priority = 100.0
+
+    def __new__(cls, *args, **kwargs):
+        if cls == Differentiable:
+            assert len(args) == 1
+            expr = args[0]
+            new_obj = sympy.Expr.__new__(cls, expr)
```

Comment:
out of curiosity: what happens if you rather do:
```sympy.Expr.__new__(sympy.Expr, expr)```
and then
```new_obj._space_order = ...
new_obj._dtype = ...
...
return new_obj
```
do we get "true" SymPy objects but with additional attributes? or do you actually get exceptions when trying to add new fields?

Reply:
most likely an exception . Scratch this.